controllers/machine_controller_noderef.go

```diff
@@ -85,6 +86,22 @@ func (r *MachineReconciler) reconcileNode(ctx context.Context, cluster *clusterv
 		r.recorder.Event(machine, corev1.EventTypeNormal, "SuccessfulSetNodeRef", machine.Status.NodeRef.Name)
 	}
 
+	// Reconcile node annotations.
+	desired := map[string]string{
+		clusterv1.ClusterNameAnnotation:      machine.Spec.ClusterName,
+		clusterv1.ClusterNamespaceAnnotation: machine.GetNamespace(),
+		clusterv1.MachineAnnotation:          machine.Name,
+	}
+	if owner, err := util.GetMachineOwner(ctx, r.Client, machine); err == nil && owner != nil {
+		desired[clusterv1.OwnerKindAnnotation] = owner.GetKind()
+		desired[clusterv1.OwnerNameAnnotation] = owner.GetName()
+	}
```

Comment:
I don't think we actually need to go this far, I think we could do something like this instead:

```suggestion
	if owner := metav1.GetControllerOfNoCopy(obj); owner != nil {
		desired[clusterv1.OwnerKindAnnotation] = owner.Kind
		desired[clusterv1.OwnerNameAnnotation] = owner.Name
	}
```

Reply:
True, this works and is a lot simpler 😅 thanks!